test/token-pool-proxy.test.js

```diff
@@ -53,4 +94,156 @@ contract('Token Pool', accounts => {
       expect(tokenAddr).to.be.equal(ctx.contracts.token.address)
     })
   })
+
+  describe('sets token pool reward', () => {
+    it("doesn't set 0", async () => {
+      const tx = ctx.contracts.pool.setTokenPoolReward(0, {from: ctx.actors.mfg})
+      await errors.expectRevert(tx)
+    })
+    it('sets new reward', async () => {
+      const tx = await ctx.contracts.pool.setTokenPoolReward(10, {from: ctx.actors.mfg})
+
+      expect(tx.logs.length).to.be.equal(1)
+      const log = tx.logs[0]
+      expect(log.event).to.be.equal('TokenPoolRewardUpdated')
+    })
+    it('does not set same reward as previously set', async () => {
+      await ctx.contracts.pool.setTokenPoolReward(10, {from: ctx.actors.mfg})
+
+      const tx = ctx.contracts.pool.setTokenPoolReward(10, {from: ctx.actors.mfg})
+      await errors.expectRevert(tx)
+    })
+  })
+
+  describe('change manufacturer wallet', () => {
+    it('new address cannot be 0x0', async () => {
+      const fn = ctx.contracts.pool.changeManufacturerWallet(0x0, {from: ctx.actors.mfg})
+      await errors.expectRevert(fn)
+    })
+    it('must be a manufacturer', async () => {
+      await ctx.contracts.member.addNetworkMember(ctx.actors.admin, false, false, false, 'test_pools', {from: ctx.actors.owner})
+
+      const fn = ctx.contracts.pool.changeManufacturerWallet(ctx.actors.admin, {from: ctx.actors.alice})
+      await errors.expectRevert(fn)
+    })
+    it('not already an admin', async () => {
+      await ctx.contracts.member.addNetworkMember(ctx.actors.admin, true, false, false, 'test_pools', {from: ctx.actors.owner})
+
+      const fn = ctx.contracts.pool.changeManufacturerWallet(ctx.actors.admin, {from: ctx.actors.mfg})
+      await errors.expectRevert(fn)
+    })
+    it('not already an manufacturer', async () => {
+      await ctx.contracts.member.addNetworkMember(ctx.actors.admin, false, true, false, 'test_pools', {from: ctx.actors.owner})
+
+      const fn = ctx.contracts.pool.changeManufacturerWallet(ctx.actors.admin, {from: ctx.actors.mfg})
+      await errors.expectRevert(fn)
+    })
+    it('not already an irn NODE', async () => {
+      await ctx.contracts.member.addNetworkMember(ctx.actors.admin, false, false, true, 'test_pools', {from: ctx.actors.owner})
+
+      const fn = ctx.contracts.pool.changeManufacturerWallet(ctx.actors.admin, {from: ctx.actors.mfg})
+      await errors.expectRevert(fn)
+    })
+    it('token pool does not already exist', async () => {
+      ctx.contracts.member.addNetworkMember(ctx.actors.alice, false, false, false, 'test_pools', {from: ctx.actors.owner})
```

Comment:
this needs to setup a new admin at test_pools_new, and do a single registration or depsit some tokens so that it has a balance, then the test will hit the 'pool already exists' require.  This is a bug we had in original tests, so good catch now :)

Reply:
for this test we don't actually need to transfer tokens, right? we can just set the balance in storage?

Reply:
We can also do the token transfer, those are two diff tests

Reply:
added strorage write, will need to check coverage reports to make sure require is hit.